lib/LaTeXML/Util/Pathname.pm

```diff
@@ -315,8 +315,18 @@ sub pathname_find {
 sub pathname_findall {
   my ($pathname, %options) = @_;
   return unless $pathname;
-  my @paths = candidate_pathnames($pathname, %options);
-  return grep { -f $_ } @paths; }
+  my @paths = grep { -f $_ } candidate_pathnames($pathname, %options);
+  if (!@paths) {    # Fallback:
+     # To allow arbitrarily cased spellings,
+     # lowercase the target path and all candidate paths
+    $pathname = lc($pathname);
+    my %star_candidates = ();
+    # This is a bit of a slow call, but really quite OK to run in rare fallback cases
+    # an average arXiv article may return with ~50-100 pathnames when looking for '*' graphics
+    for my $c (candidate_pathnames('*', %options)) {
```

Comment:
hmm....
Actually, as far as the name & extension go the deferred-case-insensitivity could actually be incorporated  into `candidate_pathnames` without any extra fs calls; In the last double loop over dirs and exts, you could (eg) dup each `push(@paths,...` with a `push(@lcpaths,...` that matches case insensitively.  Then at the end, `return(@paths ? @paths : @lcpaths);`.

What that doesn't handle --- but neither does your version! --- is case mismatch with search path directories, or a directory component on the given `$pathname`. But then, w/o testing, not sure whether it even should?  (except probably the latter)

Reply:
Right, let's check the last part. For including `{maybe/illu_blur_mnist.pdf}` in a directory I named `mAybe` I see:

```
LaTeX Warning: File `mAybe/illu_blur_MNIST.pdf' not found on input line 6.
```

So I think the directory components are not included in the casing guards (thankfully)

Reply:
I'm a bit confused about the idea to move it down to `candidate_pathnames`. Wouldn't that imply *always* computing the fallback list? The location I added it at first checks that a non-empty list of candidates has no actual files (`-f`) on the file system, and only then tries to figure out alternatives. The earliest time we know to need the fallback for certain is after the first batch of file system checks - otherwise we'll always piggyback the files?

Reply:
Specifically in my example, the regular/master `candidate_pathnames` come back as:
```
/tmp/maybe/illu_blur_MNIST
/tmp/maybe/illu_blur_MNIST.svg
/tmp/maybe/illu_blur_MNIST.png
/tmp/maybe/illu_blur_MNIST.gif
/tmp/maybe/illu_blur_MNIST.jpg
/tmp/maybe/illu_blur_MNIST.jpeg
/tmp/maybe/illu_blur_MNIST.eps
/tmp/maybe/illu_blur_MNIST.ps
/tmp/maybe/illu_blur_MNIST.postscript
/tmp/maybe/illu_blur_MNIST.ai
/tmp/maybe/illu_blur_MNIST.pdf
/tmp/maybe/illu_blur_MNIST.SVG
/tmp/maybe/illu_blur_MNIST.PNG
/tmp/maybe/illu_blur_MNIST.GIF
/tmp/maybe/illu_blur_MNIST.JPG
/tmp/maybe/illu_blur_MNIST.JPEG
/tmp/maybe/illu_blur_MNIST.EPS
/tmp/maybe/illu_blur_MNIST.PS
/tmp/maybe/illu_blur_MNIST.POSTSCRIPT
/tmp/maybe/illu_blur_MNIST.AI
/tmp/maybe/illu_blur_MNIST.PDF
```

which is very specific to the cased `illu_blur_MNIST` and doesn't need any filesystem checks to be constructed and returned back as `@paths`. Then the `-f` checks whittle it down to the empty array.

 However, if the actual file was named `/tmp/maybe/iLLu_blur_MNIST.PDF` or `/tmp/maybe/illu_blur_MnisT.PDF` etc, there would be no way to know what string to try for efficiently - I'd need to permute all possible cased namings. Instead, going for the `/tmp/maybe/*.pdf`, `/tmp/maybe/*.PDF` , ... variants via `readdir()` in `candidate_pathnames`, and then intersecting with the lowercased variant, ought to be a reasonable way of implementing things.

Since the star variants require a `readdir()`, which is a file-system operation, I think we should really only run it when we are sure none of the usual literal matches worked. Or did I misunderstand your suggestion?